### DISCUSSION TITLE
Multiple seaweed servers (in different regions) accessing the same s3 bucket through cloud drive?

### DISCUSSION THREAD

**Discussion by lewisl9029** [seaweedfs/seaweedfs#2984] [Q&A] [answered]
Hi there!

I've been running a single seaweedfs server in filer mode in a single region on https://fly.io/ until now, and I've been super happy with the performance so far. Lately I've been thinking about how to turn this into a more production-ready, multi-region setup with better data availability guarantees. 

My favorite idea so far is to spin up a seaweed server in each fly region in [Cloud Drive](https://github.com/chrislusf/seaweedfs/wiki/Cloud-Drive-Architecture) mode, with all of them pointing to the same s3 bucket for sync and write-back.

I'm thinking this should enable super fast writes and cached reads in every region, with the s3 bucket as the single source of truth and backup, without the additional operational complexity of having to plan out a good topology for active-active replication and evolving it over time, dealing with split brain scenarios, etc.

Everything I plan to store in seaweed is going to be content-addressed (hashes as file names) and immutable, so I'm thinking that should make conflicting write-backs between regions a non-issue. But I'd love to hear from more folks on any other potential issues with this setup that I might be missing!

Thanks!

**Comment by lewisl9029** [comment]
Also, a few additional questions to make sure I fully understand how Cloud Drive would work in this setup:

1. When I try to GET a file that isn't yet locally cached (both metadata and content are missing), will it attempt to fetch from s3 before returning a 404? If not, would it be possible to configure Seaweed to do that? This would allow me to automatically keep the cache fully up-to-date, while caching only the data actually being requested in each region, without having to rely on periodic syncs that can still result in stale 404s (and since everything is content-addressed and immutable, I don't need to worry about updates/deletes). I'm totally new to Go, but would be happy to try to start a PR for this if it sounds like a good feature for Seaweed.

2. In the mean time, assuming 1) is not possible yet, what is the shortest safe amount of time to wait between each `remote.meta.sync` and `remote.cache` to keep the local cache up to date as much as possible, without running into potential issues?

3. Should I be worried about race conditions between the `remote.meta.sync` commands and the write-back process `filer.remote.sync`? Say if I write a file that gets saved locally, and then issue a `remote.meta.sync` before seaweed has had chance to write back to s3 (is this even possible?), could the file be lost?

**Comment by chrislusf** [comment, marked as answer]
If metadata is missing, the file will not be found. So it depends how frequently you run `remote.meta.sync`.

Another way is to have a callback mechanism with https://docs.aws.amazon.com/AmazonS3/latest/userguide/NotificationHowTo.html
By subscribing to AWS S3 events, the metadata can be updated. This is not done yet.

**Reply by lewisl9029** [reply]
Ah, got it. Should I be worried about running `remote.meta.sync` _too_ often? I.e. is it safe to run it every 1 minute, 10 seconds, 5 seconds, 1 second?
